keymanager/src/runtime/methods.rs

```diff
@@ -62,14 +63,18 @@ pub fn get_or_create_ephemeral_keys(
 pub fn get_public_ephemeral_key(
     req: &EphemeralKeyRequest,
     ctx: &mut RpcContext,
-) -> Result<Option<SignedPublicKey>> {
+) -> Result<SignedPublicKey> {
     // No authentication or authorization.
     // Absolutely anyone is allowed to query public ephemeral keys.
     validate_epoch(req.epoch, ctx)?;
 
     let kdf = Kdf::global();
     let pk = kdf.get_public_key(req)?;
-    pk.map_or(Ok(None), |pk| Ok(Some(kdf.sign_public_key(pk)?)))
+
+    let req = cbor::to_vec((*req).clone());
+    let sig = kdf.sign_public_key(pk, req)?;
```

Comment:
drive by here.

that we're signing req along w/ pk is good but should there be more context, such as cacheability and cache expiry?  esp with ephemeral keys but probably also relevant to `get_public_key` above, once a key is no longer valid the holder of the corresponding private key could disclose it (esp if the protocols using the keypair is forward secure), but a cacheable signed statement that's unscoped saying the key is good would make that dangerous.

Reply:
I think we already have an implicit cache expiry. To verify the signature one needs RAK which should be verified with a quote and a policy. And quotes have expiration date already set.